### DISCUSSION TITLE
Public Saleor Marketplace

### DISCUSSION THREAD

**Discussion by manhtd98** [saleor/saleor#10347] [General]
### What I'm trying to achieve
I think a public library, the application is the main feature of Saleor.
Does anyone try to develop it in the soon time?
### Describe a proposed solution
Marketplace where i can choose application to install

**Comment by patrys** [comment]
Great minds think alike. We are already working on building a marketplace that will allow you to install prebuilt apps and to deploy ready to use app templates to Vercel!

**Reply by manhtd98** [reply]
do you have some application already can using. it's hard to find it on github?

**Reply by leandrobortolotto** [reply]
Do you have an estimated timeframe for the marketplace feature? I'm choosing a platform and Saleor looks excellent, but I do need the marketplace feature.

**Reply by peelar** [reply]
The first part of Saleor Cloud Marketplace: App Templates Gallery [can already be seen in the demo](https://demo.saleor.io/dashboard/marketplace/) 🕺. It's currently in feature preview, so please expect it to change, but, as you can see, we are not far from officially releasing it.

**Reply by leandrobortolotto** [reply]
Thank you @peelar. But I meant to have the possibility of Saleor running as a multivendor website. I apologize for my mistake, I misunderstood the "theme/subject" of this question/discussion in the middle of my extensive and tiring search for a multivendor that I could extend and add my own features.

Thank you, anyway.

**Reply by timuric** [reply]
@leandrobortolotto you could use saleor to build a multivendor store and create admin service that would manage permission logic. We are adding more granularity to our permissions, which would be helpful.

If you have specific multivendor feature in mind please share it with us.

**Reply by leandrobortolotto** [reply]
Hi @timuric. I have a very specific use case that I would like to achieve. Although it's specific, the multivendor functionality should be easily achieved by e-commerce software. I'll describe what I have in mind and then you see if it makes sense for you guys to consider it as a possibility to add those features to Saleor or to make it easier for me to create an App/Extension/Plug-in to customize these details.

**Reply by leandrobortolotto** [reply]
@timuric So I’m choosing an e-commerce platform to develop a Marketplace.
 
I already have an application using Vue.js on the Frontend and pure Java in the backend (which is linked to my proprietary web SaaS - nothing to do with this e-comm) and I would like to add a headless e-commerce solution to this website. The front end will be completely developed by our team in Vue.js.
 
This marketplace will have Partners/Vendors (which will have stores there) who will sell through this website. They will upload their brand and product details, such as logo, description, product photos, product descriptions, etc.
 
**IMPORTANT**: My SaaS offer 4 different plans: Free, Monthly, Yearly, or Business, plus a regular visitor/guest. And my partners will give different perks to my clients, depending on my client’s plan.
 
These 5 profiles (Guest, Free, Monthly, Yearly, or Business) will see different prices and different bonuses. For example. Prices from Partner A, Product X for:

- Guest: see normal price, $ 100.
- Free: see normal price, $ 100 and it gets a PDF as a perk.
- Monthly: see normal price + price with a discount, $100 (-10%) = $ 90.
- Yearly: see normal price + price with a discount, $100 (-30%) = $ 70.

QUESTION: Can I use the open-source solution and easily have the combination price/bonus/user type in a marketplace?
 
Plus, can I adapt the Basket Preview and Checkout page to show to different users different outcomes? I.e., to a Guest user, the system would show that if he/she chooses the Yearly plan he/she would get a $ 32 discount + 30min one-on-one call with the expert valued at $ 150. Something like that…
 
The shipment/delivery of the products (digital or physical) will be done by each vendor. When a client buys in my website/marketplace, the system will send a webhook to my CRM so I can save that info there, and send the emails to each partner.
 
The payment would be handled by Stripe Connect, so we would split the total between my company and our partners.

Probably, the e-commerce backend will become the "core" of my application, and I'll build an entire ecosystem to serve my clients around it. Including a new version of my SaaS, selling tickets to events, and who knows what else. So, I like the modularity of Saleor that allows me to expand the solution, but I do need the core of a marketplace covered first.

These are my thoughts at this moment. I'm not sure where it will take me, but these are my current needs.
